### PR TITLE
Updated versions for Storage March release

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 12.3.0-beta.1 (Unreleased)
+## 12.3.0 (2020-03-11)
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-share/CHANGELOG.md]: updates to release notes in another PR that is yet to be merged

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Release History
 
-## 12.0.0-beta.13 (Unreleased)
+## 12.0.1 (2020-03-11)
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: still yet to update a few things depending on small changes that may need to go in

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: Datalake is 12.0.1 to allow it to show up higher than the preview releases that were done for 12.0.0

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: @JimSuplizio just wanted to make sure this was the right thing to do

File: sdk/storage/azure-storage-blob-batch/CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Release History
 
-## 12.4.0-beta.2 (Unreleased)
-
+## 12.4.0 (2020-03-11)
+- Updated `azure-storage-blob` to version 12.5.0
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob-batch/CHANGELOG.md]: This should also include Alan's bug fix

File: sdk/storage/azure-storage-common/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 12.5.0-beta.1 (Unreleased)
+## 12.5.0 (2020-03-11)
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-common/CHANGELOG.md]: Why are we releasing if there are no changes?

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md (hunk 2 above)
Comment (JimSuplizio) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: @gapra-msft yes, being that datalake had released "-preview" libraries at 12.0.0, this was the right call.

**Pull request merged**